File: src/main/java/dev/deriou/airesume/entity/Job.java
package dev.deriou.airesume.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.time.LocalDateTime;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@TableName("job")
public class Job {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long enterpriseId;

    private String title;

    private String jdContent;

    private String techStack;

    private String location;

    private String status;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}

File: web/src/types/job.ts
import type { JobStatus } from './api'

export interface JobVO {
  id: number
  enterpriseId: number
  title: string
  jdContent: string
  techStack: string
  location: string
  status: JobStatus
  createdAt: string
  updatedAt: string
}

export interface JobCreateRequest {
  title: string
  jdContent: string
  techStack?: string
  location?: string
}

export type JobUpdateRequest = JobCreateRequest

export interface HotJobVO {
  jobId: number
  title: string
  enterpriseId: number
  location: string
  techStack: string
  applicationCount: number
}

export interface HotCompanyVO {
  enterpriseId: number
  enterpriseName: string
  openJobCount: number
  applicationCount: number
}
